packages/kbn-esql-validation-autocomplete/src/validation/validation.test.ts

```diff
@@ -2447,30 +2450,30 @@ describe('validation logic', () => {
       });
 
       describe('date_format', () => {
-        testErrorsAndWarnings('row var = date_format(now(), "a")', []);
-        testErrorsAndWarnings('row date_format(now(), "a")', []);
-        testErrorsAndWarnings('from a_index | eval var = date_format(dateField, stringField)', []);
-        testErrorsAndWarnings('from a_index | eval date_format(dateField, stringField)', []);
+        testErrorsAndWarnings('row var = date_format("a", now())', []);
+        testErrorsAndWarnings('row date_format("a", now())', []);
+        testErrorsAndWarnings('from a_index | eval var = date_format(stringField, dateField)', []);
+        testErrorsAndWarnings('from a_index | eval date_format(stringField, dateField)', []);
 
         testErrorsAndWarnings(
-          'from a_index | eval var = date_format(to_datetime(stringField), to_string(stringField))',
+          'from a_index | eval var = date_format(to_string(stringField), to_datetime(stringField))',
           []
         );
 
         testErrorsAndWarnings('from a_index | eval date_format(stringField, numberField)', [
-          'Argument of [date_format] must be [date], found value [stringField] type [string]',
-          'Argument of [date_format] must be [string], found value [numberField] type [number]',
+          'Argument of [date_format] must be [date], found value [numberField] type [number]',
         ]);
 
-        testErrorsAndWarnings('from a_index | eval date_format(dateField, stringField, extraArg)', [
+        testErrorsAndWarnings('from a_index | eval date_format(stringField, dateField, extraArg)', [
           'Error: [date_format] function expects no more than 2 arguments, got 3.',
         ]);
 
-        testErrorsAndWarnings('from a_index | sort date_format(dateField, stringField)', []);
+        testErrorsAndWarnings('from a_index | sort date_format(stringField, dateField)', []);
       });
 
       describe('date_parse', () => {
         testErrorsAndWarnings('row var = date_parse("a", "a")', []);
+        testErrorsAndWarnings('row var = date_parse("a")', []);
```

Comment:
Our old definition for `date_parse` inaccurately required 2 arguments when 1 is just fine.